webapp/channels/src/components/file_preview_modal/image_preview.tsx

```diff
@@ -27,22 +72,157 @@ export default function ImagePreview({fileInfo, canDownloadFiles}: Props) {
         previewUrl = fileInfo.has_preview_image ? getFilePreviewUrl(fileInfo.id) : fileUrl;
     }
 
+    const computeScaleOptions = React.useCallback(() => {
+        const imgEl = imageRef.current;
+
+        if (imgEl) {
+            setBaseScale(imgEl.naturalWidth / imgEl.clientWidth);
+            setScaleOptions([
+                {
+                    key: 'auto',
+                    value: 1,
+                    label: formatMessage({id: 'file_preview_modal_image_zoom_actions.select.auto', defaultMessage: 'Automatic zoom'}),
+                },
+                {
+                    key: 'actual',
+                    value: Math.round((imgEl.naturalWidth / imgEl.clientWidth) * 1000) / 1000, // round to 3 decimal places to prevent a duplicate value conflict with 100%
+                    label: formatMessage({id: 'file_preview_modal_image_zoom_actions.select.actual_size', defaultMessage: 'Actual size'}),
+                },
+                ...[50, 75, 100, 125, 150, 200, 300, 400].map((level) => ({
+                    key: `${level}%`,
+                    value: Math.round((imgEl.naturalWidth / imgEl.clientWidth) * level) / 100,
```

Comment:
I'm a bit surprised to see that these values are based on the image's aspect ratio instead of being constant values like `1` or `100` for 100%. Is there a particular reason you did it this way? It's a bit odd for me to see `1` being the automatic size above instead of that being some special value

Reply:
So yes for the image you would expect a scale of 1 to be the full size image, but for the zoom library a scale of 1 is the image in it's starting scale, where it is filling the container. So there is essentially a mis-match in the scales.

So you either need to let 1 be the scale of the image in it's container and then derive all the other image zoom levels from that (which is what I've done), or you let 1 be the scale of the original image, and convert the scale values going in to and coming out of the library to that scale.

My intuition was that if you convert the image scale to use the library scale then you can do the conversion math up-front (though the option values need to be re-computed when the screen size changes, which is awkward), whereas if you go the other way you need to do some conversion math every time the scale is changed.

What do you think?

Reply:
Ah, I see. This makes sense to me then. When I saw you had a state for `baseScale` as well, I thought that was being passed to the library to set what `1` means, not that you were holding onto it to do this sort of math later on. Thanks for the explanation!